<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Variance heterogeneity analysis</title>
    <meta charset="utf-8" />
    <meta name="author" content="Gota Morota  http://morotalab.org/" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Variance heterogeneity analysis
## Quantitative Genetics Short Course <span class="citation">@UFV</span>
### Gota Morota <br /><a href="http://morotalab.org/" class="uri">http://morotalab.org/</a> <br />
### 2019/11/26

---









# Epistasis what is it?
![](Cordell.png)



---
# How to parameterize response variable `\(\mathbf{y}\)` ?
Example: 50,000 SNP panel 

- additive model 
    - 50K effects to estimate
- additive + dominance model 
    - 50K + 50K effects to estimate!
- additive + dominance  + second-order epistasis model 
    - 50K + 50K RV + `\({50\text{K} \choose 2}\)` effects to estimate


1. Overparameterization!!!!!

2. Curse of dimensionality


---
# Mean heterogeneity analysis
- mean value difference between genotypes
- average effect of allele substitution
- standard GWAS is designed to capture these genetic signals

&lt;img src="mGWAS.png" height="406px" width="650px"/&gt;




---
# Variance heterogeneity analysis
- heterogenity of variance between genotypes
- variance heterogeneity loci (vQTL) or variance GWAS
(vGWAS)
- standard GWAS is not able to capture these genetic signals

&lt;img src="vGWAS.png" height="406px" width="650px"/&gt;




---
# Example: Variance heterogeneity locus
&lt;img src="varQTL.png" height="406px" width="650px"/&gt;



---
# Variance heterogeneity loci
Variance heterogeneity loci are potentially involved in
- buffer the impact of cryptic genetic variations
- release of hidden genetic variation
- developmental stability
- phenotypic plasticity


.pull-left[
&lt;img src="LynchWalsh.jpg" height="256px" width="250px"/&gt;
]


.pull-right[
&lt;img src="WalshLynch.jpg" height="256px" width="250px"/&gt;
]

--

Epistasis




---
# Non-epistatic variance QTL vs. Epistatic variance QTL 
&lt;img src="PGRP.png" height="406px" width="650px"/&gt;


---
# Association mapping panel
- hard winter wheat association panel of 299 lines
- cadmium concentration
- 90,000 markers (Wang et al. 2014)

![Guittieri et al., 2015](Guittieri2015.png)

Guittieri et al., 2015

---
# Double generalized linear model (DGLM)
$$
`\begin{align}
  \mathbf{y} &amp;= \mathbf{1}\mu_m + \mathbf{X}\mathbf{b} +  \mathbf{S}_m a_m + \boldsymbol{\epsilon} \\
   \boldsymbol{\epsilon}_i  \sim&amp; N(0, \sigma^{2}_{\epsilon_i}) \\\
    \log  (\sigma^2_{\epsilon_i}) &amp;=   \mathbf{1}\mu_v + \mathbf{S}_v a_v. 
\end{align}`
$$
- `\(\sigma^2_{\epsilon_{i}} \sim \frac{ \hat{\epsilon}_i^2 }{ 1 - h_{ii}}\)` is fitted using  GLM of a gamma distribution with a log link function, where weights are provided by `\(\frac{1 - h_{ii}}{2}\)` and a linear predictor equals to a single marker covariate `\(\mathbf{s}\mathbf{a}\)`. 

- equivalent to modeling  `\(\mathbf{y} \sim N(\mathbf{1}\mu_m + \mathbf{X}\mathbf{b} + \mathbf{S}_m a_m, \exp(\mathbf{1}\mu_v + \mathbf{S}_va_v) )\)`

- works in an iterative manner between the mean and variance parts of the model


---
# Hierarchical generalized linear model (HGLM)
$$
`\begin{align}
  \mathbf{y} &amp;= \mathbf{1}\mu_m + \mathbf{X}\mathbf{b} +  \mathbf{S}_m a_m +  \mathbf{Z}\mathbf{u} + \boldsymbol{\epsilon} 
\end{align}`
$$


- `\(\mathbf{u} \sim N(0, G \sigma^2_g)\)` is assumed to be a stochastic Gaussian process with mean zero and covariance function defined by a genomic relationship matrix `\(\mathbf{G}\)` 

Variance GWAS
- equivalent to peforming vGWAS after adjusting for  potential covariates and mean-controlling effect

---
# Epistasis analysis
Pairwise interaction analyses for significant variance heterogenity markers 
$$
`\begin{align*}
\mathbf{y} &amp;= \mathbf{1}\mu +  \mathbf{X}\mathbf{b} +  \mathbf{S}_j\mathbf{a}_j + \mathbf{S}_k\mathbf{a}_k + ( \mathbf{S}_j \mathbf{S}_k)\nu_{jk} + \boldsymbol{\epsilon}
\end{align*}`
$$

- `\(\mathbf{X}\)` is the incident matrix for the first four PCs

- `\(S_j\)` and `\(S_k\)` are SNP codes for the jth and kth markers


---
# Results: Mean and variance GWAS
&lt;img src="circle1.png" height="506px" width="600px"/&gt;

---
# Results: Variance GWAS loci
&lt;img src="violin1.png" height="506px" width="600px"/&gt;

---
# Results: Epistasis analysis
&lt;img src="epi1.png" height="506px" width="600px"/&gt;

---
# Preprint
![](bioRxiv.png)


---
# Morota lab at Virginia Tech
![](SabrinaParty.jpg)
&lt;img src="SabrinaParty.jpg" height="506px" width="600px"/&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
